lib/services/device.dart

```diff
@@ -9,34 +9,37 @@ import 'package:shared_preferences/shared_preferences.dart';
 
 enum NotificationType { RESUME, PAUSE }
 
-class Device {
+class Device extends ClipboardListener {
   static const EventChannel _notificationsChannel =
       EventChannel('com.breez.client/lifecycle_events_notifications');
-  static const MethodChannel _breezShareChannel =
-      MethodChannel('com.breez.client/share_breez');
 
   final StreamController _eventsController =
       StreamController<NotificationType>.broadcast();
   Stream<NotificationType> get eventStream => _eventsController.stream;
 
-  final _distinctClipboardController = BehaviorSubject<String>();
-  Stream get distinctClipboardStream => _distinctClipboardController.stream;
-
-  final _rawClipboardController = BehaviorSubject<String>();
-  Stream<String> get rawClipboardStream => _rawClipboardController.stream;
+  final _clipboardController = BehaviorSubject<String>();
+  Stream<String> get clipboardStream => _clipboardController.stream.where((e) => e != _lastFromAppClip);
 
-  String _lastClipping = "";
   static const String LAST_CLIPPING_PREFERENCES_KEY = "lastClipping";
+  static const String LAST_FROM_APP_CLIPPING_PREFERENCES_KEY = "lastFromAppClipping";
+
+  String _lastFromAppClip;
 
   Device() {
+    log.finest("Initing Device");
+
     var sharedPreferences = SharedPreferences.getInstance();
     sharedPreferences.then((preferences) {
-      _lastClipping =
-          preferences.getString(LAST_CLIPPING_PREFERENCES_KEY) ?? "";
+      _lastFromAppClip = preferences.getString(LAST_FROM_APP_CLIPPING_PREFERENCES_KEY);
+      _clipboardController.add(preferences.getString(LAST_CLIPPING_PREFERENCES_KEY) ?? "");
```

Comment:
@ademar111190  Why do we add to the clipboard stream the last clipped value from the persistent storage?

Reply:
That is a great question, I haven't noticed before, but we always skip 1 because of that; I'll remove this logic because for me it looks like a residual thing from the previous clipboard logic.

Reply:
I just removed it, but I'll add it back, I did find a flow where it is necessary; If you close the app and open it again with the same clipboard content you are going to trigger a flow that you already ran o the previous session.

Reply:
I did add back, now that specific flow works as expected too

Reply:
Could you add comments for that case

> I just removed it, but I'll add it back, I did find a flow where it is necessary; If you close the app and open it again with the same clipboard content you are going to trigger a flow that you already ran o the previous session.

Could you add this case as a comment above skip logic?

Reply:
Comment added